pkg/operator/ceph/file/controller.go

```diff
@@ -164,7 +164,7 @@ func (c *FilesystemController) onUpdate(oldObj, newObj interface{}) {
 // ParentClusterChanged determines wether or not a CR update has been sent
 func (c *FilesystemController) ParentClusterChanged(cluster cephv1.ClusterSpec, clusterInfo *cephconfig.ClusterInfo, isUpgrade bool) {
 	c.clusterInfo = clusterInfo
-	if !isUpgrade {
+	if cluster.CephVersion.Image == c.clusterSpec.CephVersion.Image {
```

Comment:
Not really because `isUpgrade` is computed way earlier and detects whether or not this is a Ceph upgrade. You shouldn't worry too much as this code might go away soon.

Reply:
Ok. I'm still seeing an inconsistency here, maybe cosmetic, maybe it's just my lack of familiarity with the rook code: Yes, `isUpgrade` was calculated earlier, and it was handed into the function, and was previously used to decide whether to take the upgrade path here in `ParentClusterChanged`. Now we are replacing the entity used to take the decision by a re-calculated expression. That tells me that the previously calculated `isUpgrade` was not good enough (otherwise you would not need this change here). But instead of using the presumably more accurate re-calculated expression, you use `isUpgrade` later in the function.

So my confusion is: either the previously calculated `isUpgrade` is correct, then we should not need this hunk. Or it is not correct, then we should not be using `isUpgrade` or replace its earlier calculation.

I really don't want to slow down the review of this, but this just struck me...

Reply:
Ok, so basically the `if !isUpgrade {` was completely wrong because `isUpgrade` tells us if the new image that was injected has a newer version of Ceph or not.
They are other things that we do incorrectly, one of them if to only run the upgrade checks only when `isUpgrade` is True, we should do that all the time, it's tracked here: https://github.com/rook/rook/issues/4379.
I just don't want to fix it now, because the change might be bigger and I don't want to touch too much code at the moment.

Now, doing `if cluster.CephVersion.Image == c.clusterSpec.CephVersion.Image {` tells us that the image from the spec did not change and that there is nothing to do on the deployments. So that the code returns more rapidly.

So for future work, `isUpgrade` should be modified.

Does that help?

Reply:
Yeah, thanks!